articles/151_roslaunch_xml.md

```diff
@@ -0,0 +1,744 @@
+      <xs:annotation>
+        <xs:documentation xml:lang="en">
+          Name of the node executable.
+        </xs:documentation>
+      </xs:annotation>
+    </xs:attribute>
+    <xs:attribute name="name" type="xs:string" use="optional">
+      <xs:annotation>
+        <xs:documentation xml:lang="en">
+          A name for the launched ROS node.
+        </xs:documentation>
+      </xs:annotation> 
+    </xs:attribute>
+      <xs:attribute name="args" type="xs:string" use="optional">
+        <xs:annotation>
+          <xs:documentation xml:lang="en">
+            Additional 'command-line' arguments for the ROS node.
+          </xs:documentation>
+        </xs:annotation>
+      </xs:attribute>
+      <xs:attribute name="ns" type="xs:string" use="optional">
+        <xs:annotation>
+          <xs:documentation xml:lang="en">
+            A ROS namespace to scope the launched ROS node.
+          </xs:documentation>
+        </xs:annotation>
+      </xs:attribute>
+      <xs:attribute name="launch-prefix" type="xs:string" use="optional">
+        <xs:annotation>
+          <xs:documentation xml:lang="en">
+            A prefix for the ROS node command-line used for launch.
+          </xs:documentation>
+        </xs:annotation>
+      </xs:attribute>
+      <xs:attribute name="output" use="optional">
+        <xs:annotation>
+          <xs:documentation xml:lang="en">
+            Output type for the launched ROS node.
+          </xs:documentation>
+        </xs:annotation>
+        <xs:simpleType>
+          <xs:restriction base="xs:string">
+            <xs:enumeration value="log">
+              <xs:annotation>
+                <xs:documentation xml:lang="en">
+                  ROS node output goes to a log file.
+                </xs:documentation>
+              </xs:annotation>
+            </xs:enumeration>
+            <xs:enumeration value="screen">
+              <xs:annotation>
+                <xs:documentation xml:lang="en">
+                  ROS node output goes to stdout.
+                </xs:documentation>
+              </xs:annotation>
+            </xs:enumeration>
+          </xs:restriction>
+        </xs:simpleType>
+      </xs:attribute>
+      <xs:attribute name="if" type="xs:string" use="optional">
+        <xs:annotation>
+          <xs:documentation xml:lang="en">
+            A predicate to condition ROS node launch i.e. only do
+            so if the predicate evaluates to a truthy value.
+          </xs:documentation>
+        </xs:annotation>
+      </xs:attribute>
+      <xs:attribute name="unless" type="xs:string" use="optional">
+        <xs:annotation>
+          <xs:documentation xml:lang="en">
+            A predicate to condition ROS node launch i.e. do so unless
+            the predicate evaluates to a truthy value.
+          </xs:documentation>
+        </xs:annotation>
+      </xs:attribute>
+  </xs:element>
+</xs:schema> 
+```
```

Comment:
I'm not sure that it is good to have the full schema here, just because it will need to be kept up-to-date with an actual schema file at some point. That's not to say that the schema isn't useful, just that here might not be the best place for it.

Reply:
So while I agree in principle with not having to maintain multiple copies of it, the format document, or format specification to be precise, **is** the schema. Meaning, I don't see much value on a having format specification without a description of that format. Moreover, the rest of the document (e.g. the examples) revolves around that schema, so you'd have to keep them up to date anyways.

We can discuss where this should live though. Maybe this isn't the best place? What do you think?